src/backends/github/API.js

```diff
@@ -20,70 +21,47 @@ export default class API {
   }
 
   isCollaborator(user) {
-    return this.request('/user/repos').then((repos) => {
-      let contributor = false
-      for (const repo of repos) {
-        if (repo.full_name === this.repo && repo.permissions.push) contributor = true;
-      }
-      return contributor;
-    }).catch((error) => {
-      console.error("Problem with response of /user/repos from GitHub");
-      throw error;
-    })
+    return this.request('/user/repos').then(repos =>
+      Object.keys(repos).some(
+        key => (repos[key].full_name === this.repo) && repos[key].permissions.push
+      )
+    );
   }
 
   requestHeaders(headers = {}) {
-    const baseHeader = {
-      "Content-Type": "application/json",
+    return {
       ...headers,
+      ...(this.token ? { Authorization: `token ${ this.token }` } : {}),
+      "Content-Type": "application/json",
     };
-
-    if (this.token) {
-      baseHeader.Authorization = `token ${ this.token }`;
-      return baseHeader;
-    }
-
-    return baseHeader;
-  }
-
-  parseJsonResponse(response) {
```

Comment:
Nice fat trimmin'

Reply:
Update: the Git Gateway backend is now using this as well - maybe worth keeping and refactoring.